modules/gui/include/visp3/gui/vpColorBlindFriendlyPalette.h

```diff
@@ -0,0 +1,203 @@
+
+  /**
+   * \brief Cast a \b vpColorBlindFriendlyPalette in a vector {R, G, B}.
+   * A \b vpColorBlindFriendlyPalette::Palette::COUNT object is set as white, i.e. {255, 255, 255}.
+   * 
+   * \return std::vector<unsigned char> A vector containing the R, g, B components (in his order), expressed 
+   * as an unsigned char (so between 0 and 255).
+   */
+  std::vector<unsigned char> to_RGB() const;
+
+  /**
+   * \brief Cast the object in a vector of doubles that belong to the range [0; 1]. The
+   * initial R, G, B values are divided by 255.
+   * For instance a pixel whose RGB values would be {0; 127.5; 255.} would be transformed in
+   * a vector {0.; 0.5; 1.0}.
+   * 
+   * \return std::vector<double> R, G, B values normalized in doubles in the range [0; 1.] by 
+   * dividing the original values by 255.
+   */
+  std::vector<double> to_colorRatio() const;
+
+  /**
+   * \brief Set the fromString object
+   * 
+   * \param nameColor 
+   * \return true if a match was found and it was possible to set \b _colorID correctly.
+   * \return false if the input string \b nameColor did not match any known names.
+   */
+  bool set_fromString(const std::string& nameColor);
+
+  /**
+   * \brief Get the name of the \b vpColorBlindFriendlyPalette object.
+   * 
+   * \return std::string The name of the color.
+   */
+  std::string to_string() const;
+
+  /**
+   * \brief Cast the object into an unsigned int that matches the value of its \b _colorID member.
+   * 
+   * \return unsigned int that matches the value of its \b _colorID member.
+   */
+  unsigned int to_uint() const;
+
+  /**
+   * \brief Get the list of available colors names.
+   * 
+   * \param prefix Optional prefix that will be written before starting the list.
+   * \param separator Optional separator between each member of the list.
+   * \param suffix Optional suffix that will be written after ending the list.
+   * \return std::string The list of names of the available colors.
+   */
+  static std::string getAvailableColorsNames(const std::string &prefix = "", const std::string &separator = " ", const std::string &suffix = "" );
+private:
+  static std::vector<vpColor> g_palette; /*!< A vector that contains the \b vpColor corresponding to the \b vpColorBlindFriendlyPalette in terms of R, g, B values.*/
+
+  /**
+   * \brief Cast \b vpColorBlindFriendlyPalette::Palette enum value into an unsigned int.
+   * 
+   * \param colorID A \b vpColorBlindFriendlyPalette::Palette enum 
+   * value we are interested in knowing the value, expressed as unsigned int.
+   * \return unsigned int that matches the value of \b colorID .
+   */
+  static unsigned int to_uint(Palette colorID);
+
+  /**
+   * \brief Get the name that corresponds to \b colorID .
+   * 
+   * \param colorID A \b vpColorBlindFriendlyPalette::Palette enum 
+   * value we are interested in knowing the name.
+   * \return std::string The corresponding name.
+   */
+  static std::string to_string(Palette colorID);
+
```

Comment:
`const` ?

Reply:
Done in [0f2e5d6](https://github.com/lagadic/visp/pull/1163/commits/0f2e5d64fa6b9817dde709113bf3c7b89a569929)